File: Leetcode/PY/Leetcode36-有效的数独.py
from typing import List
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row,col = [[0] * 10 for _ in range(9)],[[0] * 10 for _ in range(9)]
        box = [[[0] * 10 for _ in range(3)] for _ in range(3)]
        for i in range(9):
            for j in range(9):
                if board[i][j] == '.':
                    continue
                t = ord(board[i][j]) - ord('0')
                row[i][t] += 1
                col[j][t] += 1
                box[i // 3][j // 3][t] += 1
                if row[i][t] > 1 or col[j][t] > 1 or box[i // 3][j // 3][t] > 1:
                    return False
        return True 

File: Leetcode/sbgo/Leetcode36-有效的数独.go
package sbgo

func isValidSudoku(board [][]byte) bool {
	var row, col [9][10]int
	var box [3][3][10]int
	for i := 0; i < 9; i++ {
		for j := 0; j < 9; j++ {
			if board[i][j] == '.' {
				continue
			}
			u := board[i][j] - '0'
			row[i][u]++
			col[j][u]++
			box[i/3][j/3][u]++
			if row[i][u] > 1 || col[j][u] > 1 || box[i/3][j/3][u] > 1 {
				return false
			}
		}
	}
	return true
}
